lldb/source/Plugins/LanguageRuntime/Swift/SwiftMetadataCache.cpp

```diff
@@ -213,25 +184,30 @@ void SwiftMetadataCache::cacheFieldDescriptors(
   }
 
   auto module = std::get<ModuleSP>(it->second);
-  // Write the data to the body encoder with the format expected by the current
-  // cache version.
-  DataEncoder body_encoder;
-  if (!writeMangledNamesAndOffsetsToEncoder(info_id, field_descriptors,
-                                            mangled_names, body_encoder))
-    return;
 
-  uint64_t typeref_buffer_size = body_encoder.GetData().size();
-  llvm::StringRef typeref_buffer((const char *)body_encoder.GetData().data(),
-                                 typeref_buffer_size);
+  auto maybe_pair =
+      generateHashTableBlob(info_id, field_descriptors, mangled_names);
+  if (!maybe_pair)
+    return;
 
-  llvm::SmallString<0> compressed_buffer;
-  llvm::zlib::compress(typeref_buffer, compressed_buffer);
+  auto &table_offset = maybe_pair->first;
+  auto &hash_table_blob = maybe_pair->second;
 
   // Write the header followed by the body.
   DataEncoder encoder;
-  encoder.AppendCString(module->GetUUID().GetAsString());
-  encoder.AppendU64(typeref_buffer_size);
-  encoder.AppendData(compressed_buffer);
+  auto uuid = module->GetUUID().GetAsString();
+  encoder.AppendCString(uuid);
```

Comment:
It's not a big deal in terms of performance, but why don't we store the UUID in binary?

Reply:
We do, the final binary is generated by the encoder. We store UUID + hash table in the binary.

Reply:
`GetUUID().GetAsString()` at least sounds is if it were encoding the uuid into ASCII instead of streaming out the 16bytes directly?

Reply:
or put another way — if it's  binary, why the null terminator?

Reply:
Oh I see what you mean now. I'll try to change that.

Reply:
Changed it to use the UUID's binary format